src/scripts/main.js

```diff
@@ -1,3 +1,256 @@
+    if (activeRow) {
+      activeRow.classList.remove('active');
+    }
+
+    row.classList.add('active');
+    activeRow = row;
+  });
+});
+
+// 3. Write a script to add a form to the document.
+// Form allows users to add new employees to the spreadsheet.
+const formHtml = `
+  <form class="new-employee-form">
+    <label>
+      Name:
+      <input
+        name="name"
+        type="text"
+        data-qa="name"
+      />
+    </label>
+    <label>
+      Position:
+      <input
+        name="position"
+        type="text"
+        data-qa="position"
+      />
+    </label>
+    <label>
+      Office:
+      <select
+        name="office"
+        data-qa="office"
+      >
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+      />
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+      />
+    </label>
+    <button
+      type="button"
+      id="save-button"
+    >
+      Save to table
+    </button>
+  </form>
+  `;
+
+table.insertAdjacentHTML('afterend', formHtml);
+
+// 4. Show notification if form data is invalid
+function pushNotification(message, type) {
+  const notification = document.createElement('div');
+  const titleSpan = document.createElement('span');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
+
+function validateFormData(formData) {
+  const nameRegex = /^[A-Za-z\s]+$/;
+
+  if (formData.name.length < 4 || !nameRegex.test(formData.name)) {
+    pushNotification(
+      'Name must be at least 4 characters long and contain only letters.',
+      'error',
+    );
+
+    return false;
+  }
+
+  if (formData.age < 18 || formData.age > 90) {
+    pushNotification('Age must be between 18 and 90.', 'error');
+
+    return false;
+  }
+
+  if (!formData.position.trim()) {
+    pushNotification('Position is required.', 'error');
+
+    return false;
+  }
+
+  if (!formData.office.trim()) {
+    pushNotification('Office is required.', 'error');
+
+    return false;
+  }
+
+  const salary = parseFloat(formData.salary);
+
+  if (isNaN(salary) || salary <= 0) {
+    pushNotification('Salary must be a valid positive number.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
```

Comment:
Code quality issue: The `validateFormData` function is doing too much. It's validating all the form fields and showing the error notifications. Consider breaking this function into smaller ones, each responsible for validating a single form field. This would make the code easier to read and maintain.